### ISSUE TITLE
Report naming scheme

### ISSUE THREAD

**Issue by tonsky** [clojure-goes-fast/clj-async-profiler#38]
Have you ever considered naming report files as

```
2024-10-14-15-47-44-cpu-flamegraph-01.html
```

instead of

```
01-cpu-flamegraph-2024-10-14-15-47-44.html
```

Then they will be sorted correctly with name sorting even between different invocations. Right now if have 01-, 02-, 03-... generated, restart my app and generate new report, it’ll have an index of 01- again

Alternative would be looking into existing files and starting counting from there.

I can provide a PR if that’s something you’d consider

**Comment by alexander-yakushev** [comment]
Hello, thanks for the feature request!

You are absolutely right; the current sort order is quite ridiculous, and I'm annoyed by it daily. I'm currently in the middle of redesigning the "index" page of the profiler UI where the flamegraphs will be sorted by date by default.

That being said, I planned to keep the order of parts in the name the same (and maybe add more "fields" in the future). If the order in the UI was sorted by date, would that be enough for you? Or do you interact with the flamegraph files as files too (e.g., `ls` them in the terminal), so date being in front of the name would still be important?

**Comment by tonsky** [comment]
Yes, I usually open the folder directly, never tried the UI. This is how I name files right now (dashes in dates removed so less noise)


![Screenshot 2024-10-15 at 12 24 21](https://github.com/user-attachments/assets/d15d17ef-9179-485d-aff2-e0d589ac6ddf)

**Comment by alexander-yakushev** [comment]
Hi, you may want to check out version `1.4.0-SNAPSHOT` that I've just pushed. I've changed the naming scheme of the files to have the datetime in front. I've also redesigned the UI so now it looks like this: 
<img width="870" alt="image" src="https://github.com/user-attachments/assets/d076f661-5ca8-48d3-be3a-5b766f22caba">
The files are sorted in time-descending order, so the newest files appear on top.

**Comment by tonsky** [comment]
Awesome! Meanwhile I started using UI to navigate results, it’s pretty convenient, I agree. Love the reverse chronological sorting!

Might I suggest making it more compact though? E.g. moving samples into its own column and chaning row height from 40 to 30?

![Screenshot 2024-10-17 at 14 52 40](https://github.com/user-attachments/assets/096a13e4-4e76-4977-8d51-f121e2571823)

**Comment by alexander-yakushev** [comment]
Thank you for the suggestion. The table now shows only the samples when it knows them, and falls back to file size if it doesn't. The filesize of the flamegraph is not a very useful metric anyway, so showing both samples and file size is unnecessary. However, the row height of 30px makes it feel quite crowded, especially as I'm adding more text and buttons. I've settled at 35px for now, we'll see how it goes.

Anyway, I've redeployed `1.4.0-SNAPSHOT` with the latest changes.

<img width="850" alt="image" src="https://github.com/user-attachments/assets/0c084d38-8c26-4b7e-bf6d-7bf2ad1944a1">
